### ISSUE TITLE
False positive for reportUndefinedVariable on Python 3.10+ using postponed evaluation of annotations

### ISSUE THREAD

**Issue by ShadowLNC** [microsoft/pyright#1900]
I think this is an issue with pyright and not pylance, sorry if I've got the wrong repo. See also: [pyflakes#637](https://github.com/PyCQA/pyflakes/issues/637) and [pylint#4490](https://github.com/PyCQA/pylint/issues/4490).

If I have the following code:

```
class X:
    def func(self) -> X:
        pass
```

Pylance reports `"X" is not defined`.

I would expect that the warning is not raised as postponed evalutation of annotations are enabled by default in Python 3.10.

If I specify `from __future__ import annotations`, it is (correctly) not raised.

I am running pyright via the pylance vscode extension `v2021.5.3`.

**Comment by ShadowLNC** [comment]
Apologies - looks like this has been deferred to 3.11
